backend/src/components/secureExchange.js

```diff
@@ -556,45 +556,80 @@ async function removeSecureExchangeStudent(req, res) {
   }
 }
 
-async function updateEdxUserRoles(req, res) {
+async function updateEdxUserSchoolRoles(req, res) {
   try {
     const token = getAccessToken(req);
     validateAccessToken(token);
     checkEDXUserAccessForSchoolAdminFunctions(req, req.body.params.schoolID);
 
-    let response = await getData(token, config.get('edx:edxUsersURL') + '/' + req.body.params.edxUserID, req.session?.correlationID);
-
-    let selectedUserSchool = response.edxUserSchools.filter(school => school.schoolID === req.body.params.schoolID);
-
-    let rolesToBeRemoved = [];
+    let edxUser = await getData(token, `${config.get('edx:edxUsersURL')}/${req.body.params.edxUserID}`, req.session?.correlationID);
+    let selectedUserSchools = edxUser.edxUserSchools.filter(school => school.schoolID === req.body.params.schoolID);
+    if (!selectedUserSchools[0]) {
+      return errorResponse(res, 'A user school entry was not found for the selected user.', HttpStatus.NOT_FOUND);
+    }
+    if (selectedUserSchools.length > 1) {
+      return errorResponse(res, 'Too many user school entries have been found for the selected user.', HttpStatus.CONFLICT);
+    }
+    let selectedUserSchool = selectedUserSchools[0];
+    let existingUserSchoolRoles = new Map(selectedUserSchool.edxUserSchoolRoles.map(edxUserSchoolRole => [edxUserSchoolRole.edxRoleCode, edxUserSchoolRole]));
 
-    //Determine roles to be removed
-    selectedUserSchool[0].edxUserSchoolRoles.forEach(function (userSchoolRole) {
-      if (!req.body.params.selectedRoles.filter(value => userSchoolRole.edxRoleCode === value).length > 0) {
-        rolesToBeRemoved.push(userSchoolRole.edxRoleCode);
+    selectedUserSchool.edxUserSchoolRoles = [];
+    req.body.params.selectedRoles.forEach(function (role) {
+      if (existingUserSchoolRoles.has(role)) {
+        selectedUserSchool.edxUserSchoolRoles.push(existingUserSchoolRoles.get(role));
+        return;
       }
+      let newRole = {};
+      newRole.edxUserSchoolID = selectedUserSchool.edxUserSchoolID;
+      newRole.edxRoleCode = role;
+      selectedUserSchool.edxUserSchoolRoles.push(newRole);
     });
 
-    selectedUserSchool[0].edxUserSchoolRoles = selectedUserSchool[0].edxUserSchoolRoles.filter(value => !rolesToBeRemoved.includes(value.edxRoleCode));
+    selectedUserSchool.updateDate = null;
+    selectedUserSchool.createDate = null;
+
+    const result = await putData(token, selectedUserSchool, `${config.get('edx:edxUsersURL')}/${selectedUserSchool.edxUserID}/school`, req.session?.correlationID);
+    return res.status(HttpStatus.OK).json(result);
+  } catch (e) {
+    log.error(e, 'updateEdxUserRoles', 'Error occurred while attempting to update user roles.');
```

Comment:
super small I think you meant updateEdxUserSchoolRoles ... update user school roles.